ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/component/StateComponent.java

```diff
@@ -41,6 +41,6 @@
      *
      * @param state the component state object
      */
-    void loadState(@NotNull JsonObject state);
+    void setState(@NotNull JsonObject state);
```

Comment:
I agree with @vparfonov, in common case `set` prefix used for setters, mutator method for object but that method doesn't has side effects. In this case we have huge side effects, so `set` prefix is confusing, `load` is better.